### PR TITLE
[WEB-12] email confirmation pages

### PR DESCRIPTION
**Pull request by UncleClapton**
This PR also does some reorganization of our file structure to keep things clean.

### CODE REVIEW

File: pages/confirmations/account-confirm.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ class Confirmation extends Component {
     Public Methods
   \***************************************************************************/
 
-  async _confirmAccount () {
+  async _confirmAccount() {
```
Comment (trezy) [inline pages/confirmations/account-confirm.js]: Nope.

File: pages/confirmations/account-confirm.js [hunk 2]
```diff
@@ -48,15 +48,15 @@ class Confirmation extends Component {
     Public Methods
   \***************************************************************************/
 
-  componentDidUpdate () {
+  componentDidUpdate() {
```
Comment (trezy) [inline pages/confirmations/account-confirm.js]: Uh-uh.

File: pages/confirmations/account-confirm.js [hunk 3]
```diff
@@ -48,15 +48,15 @@ class Confirmation extends Component {
...
     const { loggedIn } = this.props
 
     if (loggedIn) {
       Router.push('/')
     }
   }
 
-  async componentWillMount () {
+  async componentWillMount() {
```
Comment (trezy) [inline pages/confirmations/account-confirm.js]: Nein.

File: pages/confirmations/account-confirm.js [hunk 4]
```diff
@@ -66,15 +66,15 @@ class Confirmation extends Component {
     }
   }
 
-  constructor (props) {
+  constructor(props) {
```
Comment (trezy) [inline pages/confirmations/account-confirm.js]: nyet

File: pages/confirmations/account-confirm.js [hunk 5]
```diff
@@ -66,15 +66,15 @@ class Confirmation extends Component {
...
     super(props)
 
     this.state = {
       confirming: true,
     }
   }
 
-  render () {
+  render() {
```
Comment (trezy) [inline pages/confirmations/account-confirm.js]: I'm out of ways to say no.

File: pages/confirmations/email-update-confirm.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+        {(confirming && !acceptNewEmail) && (
+          <React.Fragment>
+            <p>Rolling Back...</p>
+          </React.Fragment>
+        )}
+
+        {(!confirming && error) && (
+          <React.Fragment>
+            <p><span aria-label="Sad face emoji" role="img">😞</span> Uh oh... It seems there's a problem with your confirmation code. If you're still trying to change your email, you may want to <a href="mailto:support@rollforguild.com">contact support</a>.</p>
```
Comment (trezy) [inline pages/confirmations/email-update-confirm.js]: Flip the link over to our Service Desk page.

File: store/actions/authentication.js [hunk 7]
```diff
@@ -41,6 +41,36 @@ export const confirmAccount = token => async dispatch => {
...
+export const confirmEmailUpdate = (confirmationToken, accept = true) => async dispatch => {
+  let response = null
+  let success = false
+
+  dispatch({ type: actionTypes.CONFIRM_EMAIL_UPDATE })
+
+  try {
+    const token = Cookies.get('accessToken')
+
+    response = await fetch(`/api/email/${confirmationToken}?response=${accept ? 'accept' : 'reject'}`, {
```
Comment (trezy) [inline store/actions/authentication.js]: We should be using the `apiService` rather than using `fetch` directly.

Comment (UncleClapton) [reply store/actions/authentication.js]: from what I'm seeing apiService does not do authentication yet, which is why I've been holding off.

Comment (trezy) [reply store/actions/authentication.js]: Nope, we don't actually need `withCredentials` for our local requests. That prop is only required if you're sending a CORS requests with Auth headers, but we're making requests to the same domain.

The Authorization header also [gets set automatically for us by the `Page` component](https://github.com/RollForGuild/rollforguild.com/blob/develop/components/Page.js#L94-L96).

**Comment by UncleClapton** [comment]
@trezy requested changes made.

File: .eslintrc.json [hunk 8]
```diff
@@ -29,14 +29,13 @@
     "react/react-in-jsx-scope": ["off"],
     "react/sort-comp": ["off"],
     "semi": ["error", "never"],
-    "space-before-function-paren": ["off"],
+    "space-before-function-paren": ["error", "always"],
```
Comment (trezy) [inline .eslintrc.json]: I'm pretty sure we can just remove this rule, since it's inherited from the AirBnB lint config.

Comment (UncleClapton) [reply .eslintrc.json]: AirBnB's rules are `["error", "never"]`, which remove the space between the name and the parens.

**Pull request merged**